keras/layers/recurrent.py

```diff
@@ -59,6 +67,36 @@ def _time_distributed_dense(x, w, b=None, dropout=None,
     return x
 
 
+def _zoneout(zoneout_probability, mask, new_state, old_state):
+    if 0 < zoneout_probability < 1:
+        return new_state * mask + old_state * (1. - mask)
+    else:
+        return new_state
+
+
+def _gen_dropout_mask(probability, inputs, units, training, repeats=1, is_zoneout=False):
+    if 0 < probability < 1:
+        ones = K.ones_like(K.reshape(inputs[:, 0, 0], (-1, 1)))
+        ones = K.tile(ones, (1, units))
```

Comment:
Can you make it in one step? Like (untested):

    ones = K.ones((inputs._keras_shape[0], units))